libraries/src/AWS.Lambda.Powertools.Idempotency/IdempotentAttribute.cs

```diff
@@ -124,4 +135,41 @@ protected internal sealed override T WrapSync<T>(Func<object[], T> target, objec
         var result = await idempotencyHandler.Handle();
         return result;
     }
+    
+    /// <summary>
+    /// Retrieve the payload from the annotated method parameter
+    /// </summary>
+    /// <param name="eventArgs">The <see cref="AspectEventArgs" /> instance containing the event data.</param>
+    /// <typeparam name="T"></typeparam>
+    /// <returns>The payload</returns>
+    private static JsonDocument GetPayload<T>(AspectEventArgs eventArgs)
+    {
+        JsonDocument payload = null;
+        var eventArgsMethod = eventArgs.Method;
+        var args = eventArgs.Args;
+        var isPlacedOnRequestHandler = IsPlacedOnRequestHandler(eventArgsMethod);
+        // Use the first argument if IdempotentAttribute placed on handler or number of arguments is 1
+        if (isPlacedOnRequestHandler || args.Count == 1)
+        {
+            payload = args is not null && args.Any() ? JsonDocument.Parse(JsonSerializer.Serialize(args[0])) : null;
+        }
+        else
+        {
+            //Find the first parameter in eventArgsMethod with attribute IdempotencyKeyAttribute
+            var parameter = eventArgsMethod.GetParameters().FirstOrDefault(p => p.GetCustomAttribute<IdempotencyKeyAttribute>() != null);
+            if (parameter != null)
+            {
+                // set payload to the value of the parameter
+                payload = JsonDocument.Parse(JsonSerializer.Serialize(args[Array.IndexOf(eventArgsMethod.GetParameters(), parameter)]));
+            }
+        }
+
+        return payload;
+    }
+
+    private static bool IsPlacedOnRequestHandler(MethodBase method)
+    {
+        //Check if method has two arguments and the second one is of type ILambdaContext
+        return method.GetParameters().Length == 2 && method.GetParameters()[1].ParameterType == typeof(ILambdaContext);
```

Comment:
Small note: ILambdaContext is an optional argument, which will return false on IsPlacedOnRequestHandler, but from the looks of it the if there is only one argument it is also valid.

Reply:
You got it right, either one parameter or 2 parameters where the second parameter is `ILambdaContext`